sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -74,9 +74,9 @@ private[sql] class HiveSessionCatalog(
   // | Methods and fields for interacting with HiveMetastoreCatalog |
   // ----------------------------------------------------------------
 
-  override def getDefaultDBPath(db: String): String = {
-    val defaultPath = context.hiveconf.getVar(HiveConf.ConfVars.METASTOREWAREHOUSE)
-    new Path(new Path(defaultPath), db + ".db").toString
+  override def getDatabasePath(dbName: String, path: Option[String] = None): String = {
+    val dbPath = path.getOrElse(context.hiveconf.getVar(HiveConf.ConfVars.METASTOREWAREHOUSE))
+    new Path(new Path(dbPath), dbName.toLowerCase() + ".db").toString
```

Comment:
Just checked the code in [hive metastore warehouse.java](https://github.com/apache/hive/blob/master/metastore/src/java/org/apache/hadoop/hive/metastore/Warehouse.java#L168-L172). It is much more complex than what I thought. I will try to follow what it does and rewrite this function.

CC @yhuai @andrewor14 Let me know if you think that is not necessary. Thanks!